addons/blocks2image/userscript.js

```diff
@@ -175,10 +177,18 @@ export default async function ({ addon, console, msg }) {
     let svgchild = block.svgGroup_;
     svgchild = svgchild.cloneNode(true);
     let dataShapes = svgchild.getAttribute("data-shapes");
-    svgchild.setAttribute(
-      "transform",
-      `translate(0,${dataShapes === "hat" ? "18" : "0"}) ${isExportPNG ? "scale(2)" : ""}`
-    );
+    let translateY = 0; // blocks no hat
+    const scale = isExportPNG ? 2 : 1;
+    if (dataShapes === "c-block c-1 hat") {
+      translateY = 20; // for My block
+    }
+    if (dataShapes === "hat") {
+      translateY = 16; // for Events
+      if (enabledAddons.includes("cat-blocks")) {
+        translateY += 16; // for cat ears
```

Comment:
Why doesn't this look at Blockly.BlockSvg.START_HAT_HEIGHT? Wouldn't using that remove the use for checking if cat blocks is enabled?

Reply:
I don't know mucn about blockly. I can PR that if you want.

Reply:
I there something similar for custom block hats?

Reply:
@TheColaber Send a pull request.

Reply:
`START_HAT_HEIGHT` doesn't work for custom block hats. I've fixed single stack exporting locally with `getBBox()` but export all still needs some work.

Reply:
#7783